# Preprocessing

In [1]:
import pandas as pd
import numpy as np

In [2]:
from functions import *

### Import the datasets

In [3]:
# Open the raw datasets
# Heavy Metal
hm_df = pd.read_excel('data/AllHMFiles.xlsx', engine='openpyxl').fillna(0.0)

# Pesticides
pesticides_df = pd.read_excel('data/AllPesticidesFiles.xlsx', engine='openpyxl').fillna(0.0)
classe_pesticides_df = pd.read_excel('data/pesticides.xlsx', engine='openpyxl')

# Surfaces
surfaces_df = pd.read_excel('data/absSurfs.xlsx', engine='openpyxl')

# Distances
distances_df = pd.read_excel('data/distsOneSheet.xlsx', engine='openpyxl')

In [4]:
hm_df.head(5)

REF....SUBSTANCE   PERIOD   Lead  Cadmium  Arsenic  Mercury  Copper  \
0            B0017  2017/P1  0.081     0.81      0.0      0.0     5.5   
1            B0019  2017/P1  0.062     0.36      0.0      0.0     8.5   
2            B0021  2017/P1  0.150     0.61      0.0      0.0     7.0   
3            B0023  2017/P1  0.067     0.35      0.0      0.0     6.9   
4            B0037  2017/P1  0.120     0.07      0.0      0.0     5.6   

   Chromium  Zinc  Chrome  
0     0.032    40     0.0  
1     0.029    33     0.0  
2     0.038    37     0.0  
3     0.031    37     0.0  
4     0.052    35     0.0

In [5]:
pesticides_df.head(5)

REF....SUBSTANCE   PERIOD  Aldrin  Benalaxyl  Chlorpropham  Cyanofenphos  \
0            B0017  2017/P1     0.0     0.0000          0.00           0.0   
1            B0019  2017/P1     0.0     0.0188          0.00           0.0   
2            B0021  2017/P1     0.0     0.0711          0.00           0.0   
3            B0023  2017/P1     0.0     0.0294          0.00           0.0   
4            B0037  2017/P1     0.0     0.0000          0.01           0.0   

   Cypermethrin  Ethofumesate  Fludioxonil  Fluopyram  ...  Nicosulfuron  \
0           0.0           0.0          0.0        0.0  ...           0.0   
1           0.0           0.0          0.0        0.0  ...           0.0   
2           0.0           0.0          0.0        0.0  ...           0.0   
3           0.0           0.0          0.0        0.0  ...           0.0   
4           0.0           0.0          0.0        0.0  ...           0.0   

   Piperonylbutoxide  Prosulfuron  Simazine  Terbuthylazine  Dieldrin  \
0                0.0          0.0       0.0             0.0       0.0   
1                0.0          0.0       0.0             0.0       0.0   
2                0.0          0.0       0.0             0.0       0.0   
3                0.0          0.0       0.0             0.0       0.0   
4                0.0          0.0       0.0             0.0       0.0   

   Fenamidone  Trifluralin  Cycloate  Pentachloroanisol  
0         0.0          0.0       0.0                0.0  
1         0.0          0.0       0.0                0.0  
2         0.0          0.0       0.0                0.0  
3         0.0          0.0       0.0                0.0  
4         0.0          0.0       0.0                0.0  

[5 rows x 117 columns]

In [6]:
classe_pesticides_df.head(5)

importName           name    statusEN   LMR  \
0      Aclonifen      aclonifen  authorized  0.05   
1         Aldrin         aldrin  prohibited  0.01   
2   Ametoctradin   ametoctradin  authorized  0.05   
3  Anthraquinone  anthraquinone  prohibited  0.02   
4   Azoxystrobin   azoxystrobin  authorized  0.05   

                          familyEN          typeEN  
0                   diphenyl ether       herbicide  
1                   organochlorine     insecticide  
2               triazolopyrimidine       fungicide  
3  polycyclic aromatic hydrocarbon  bird repellant  
4                      strobilurin       fungicide

In [7]:
surfaces_df.head(5)

Site  111           112           121            122  123  124  131  132  \
0  F0069  0.0  1.064122e+06  3.587087e+05       0.000000  0.0  0.0  0.0    0   
1  F0025  0.0  0.000000e+00  0.000000e+00  412752.692383  0.0  0.0  0.0    0   
2  F0027  0.0  2.741746e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
3  B0128  0.0  2.470089e+06  4.053755e+06  497496.251770  0.0  0.0  0.0    0   
4  F0023  0.0  0.000000e+00  2.559254e+05       0.000000  0.0  0.0  0.0    0   

   133  ...  411  412  421  422  423  511            512  521  522  523  
0  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
1  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
2  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
3  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
4  0.0  ...    0  0.0    0    0  0.0  0.0  284728.392578    0    0  0.0  

[5 rows x 45 columns]

In [8]:
distances_df.head(5)

Site     polyID  classCLC         dist
0  F0069   FR-88090       231  1183.022362
1  F0069  FR-131773       242   289.926324
2  F0069   FR-46167       211   221.344153
3  F0069   FR-11047       112  1038.383069
4  F0069  FR-195343       311   724.045683

### Heavy Metal

In [9]:
hm_df.head()

REF....SUBSTANCE   PERIOD   Lead  Cadmium  Arsenic  Mercury  Copper  \
0            B0017  2017/P1  0.081     0.81      0.0      0.0     5.5   
1            B0019  2017/P1  0.062     0.36      0.0      0.0     8.5   
2            B0021  2017/P1  0.150     0.61      0.0      0.0     7.0   
3            B0023  2017/P1  0.067     0.35      0.0      0.0     6.9   
4            B0037  2017/P1  0.120     0.07      0.0      0.0     5.6   

   Chromium  Zinc  Chrome  
0     0.032    40     0.0  
1     0.029    33     0.0  
2     0.038    37     0.0  
3     0.031    37     0.0  
4     0.052    35     0.0

In [10]:
MRL_HM_dict = {
    'Lead' : [1.5],
    'Cadmium' : [1],
    'Arsenic' : [0.3],
    'Mercury' : [1],
    'Copper' : [30]
}

MRL_HM = pd.DataFrame.from_dict(MRL_HM_dict)
MRL_HM

Lead  Cadmium  Arsenic  Mercury  Copper
0   1.5        1      0.3        1      30

In [11]:
hm_df['mrl_lead'] = np.where(hm_df['Lead'] > 1.5, 1, 0)
hm_df['mrl_cadmium'] = np.where(hm_df['Cadmium'] > 1, 1, 0)
hm_df['mrl_arsenic'] = np.where(hm_df['Arsenic'] > 0.3, 1, 0)
hm_df['mrl_mercury'] = np.where(hm_df['Mercury'] > 1, 1, 0)
hm_df['mrl_copper'] = np.where(hm_df['Copper'] > 30, 1, 0)

hm_df.head()

REF....SUBSTANCE   PERIOD   Lead  Cadmium  Arsenic  Mercury  Copper  \
0            B0017  2017/P1  0.081     0.81      0.0      0.0     5.5   
1            B0019  2017/P1  0.062     0.36      0.0      0.0     8.5   
2            B0021  2017/P1  0.150     0.61      0.0      0.0     7.0   
3            B0023  2017/P1  0.067     0.35      0.0      0.0     6.9   
4            B0037  2017/P1  0.120     0.07      0.0      0.0     5.6   

   Chromium  Zinc  Chrome  mrl_lead  mrl_cadmium  mrl_arsenic  mrl_mercury  \
0     0.032    40     0.0         0            0            0            0   
1     0.029    33     0.0         0            0            0            0   
2     0.038    37     0.0         0            0            0            0   
3     0.031    37     0.0         0            0            0            0   
4     0.052    35     0.0         0            0            0            0   

   mrl_copper  
0           0  
1           0  
2           0  
3           0  
4           0

In [12]:
# Dropping columns of metals without specific threshhold level defined by laws
hm_df = hm_df.drop(['Chromium'], axis=1)
hm_df = hm_df.drop(['Chrome'], axis=1)
hm_df = hm_df.drop(['Zinc'], axis=1)

# Dropping other columns not needed anymore
hm_df = hm_df.drop(['Lead'], axis=1)
hm_df = hm_df.drop(['Cadmium'], axis=1)
hm_df = hm_df.drop(['Arsenic'], axis=1)
hm_df = hm_df.drop(['Mercury'], axis=1)
hm_df = hm_df.drop(['Copper'], axis=1)

hm_df.head()

REF....SUBSTANCE   PERIOD  mrl_lead  mrl_cadmium  mrl_arsenic  mrl_mercury  \
0            B0017  2017/P1         0            0            0            0   
1            B0019  2017/P1         0            0            0            0   
2            B0021  2017/P1         0            0            0            0   
3            B0023  2017/P1         0            0            0            0   
4            B0037  2017/P1         0            0            0            0   

   mrl_copper  
0           0  
1           0  
2           0  
3           0  
4           0

In [13]:
group_hm_df = hm_df.groupby('REF....SUBSTANCE').agg(max)
group_hm_df.head()

PERIOD  mrl_lead  mrl_cadmium  mrl_arsenic  mrl_mercury  \
REF....SUBSTANCE                                                             
B0012             2020/P4         0            0            0            0   
B0017             2020/P4         0            0            0            0   
B0019             2020/P4         0            0            0            0   
B0021             2020/P4         0            1            0            0   
B0023             2020/P4         0            1            0            0   

                  mrl_copper  
REF....SUBSTANCE              
B0012                      0  
B0017                      0  
B0019                      0  
B0021                      0  
B0023                      0

### Pesticides

Fungicide :

In [14]:
classe_pesticides_df[classe_pesticides_df['typeEN'] == 'fungicide'].head(5)

importName                       name    statusEN   LMR  \
2         Ametoctradin               ametoctradin  authorized  0.05   
4         Azoxystrobin               azoxystrobin  authorized  0.05   
5            Benalaxyl                  benalaxyl  prohibited  0.05   
7  Benthiavalicarb.iso  benthiavalicarb-isopropyl  authorized  0.05   
8     Benzovindiflupyr           benzovindiflupyr  authorized  0.05   

             familyEN     typeEN  
2  triazolopyrimidine  fungicide  
4         strobilurin  fungicide  
5        phenylamides  fungicide  
7           carbamate  fungicide  
8               amide  fungicide

In [15]:
fungicides = classe_pesticides_df[classe_pesticides_df['typeEN'] == 'fungicide'].importName.to_list()
fungicides.append('Diphenylamine')
fungicides.append('Tebuconazole')
fungicides

['Ametoctradin',
 'Azoxystrobin',
 'Benalaxyl',
 'Benthiavalicarb.iso',
 'Benzovindiflupyr',
 'Boscalid',
 'Bupirimate',
 'Captan',
 'Carbendazim',
 'Chlorothalonil',
 'Cyazofamid',
 'Cyflufenamid',
 'Cymoxanil',
 'Cyprodinil',
 'Difenoconazole',
 'Dimethomorph',
 'Dimoxystrobin',
 'Dodine',
 'Epoxiconazole',
 'Ethirimol',
 'Famoxadone',
 'Fenamidone',
 'Fenbuconazole',
 'Fenhexamid',
 'Fenpropidin',
 'Fenpropimorph',
 'Fluazinam',
 'Fludioxonil',
 'Fluopicolide',
 'Fluopyram',
 'Flutolanil',
 'Fluxapyroxad',
 'Hexaconazole',
 'Iprovalicarb',
 'Isofetamid',
 'Mandipropamid',
 'Metalaxyl',
 'Metconazole',
 'Metrafenone',
 'Penconazole',
 'Pencycuron',
 'Pentachloroanisol',
 'Prochloraz',
 'Propamocarb',
 'Propiconazole',
 'Prothioconazole_d',
 'Pyraclostrobin',
 'Pyrimethanil',
 'Quinoxyfen',
 'Spiroxamine',
 'Tetraconazole',
 'Thiabendazole',
 'Thiophanate.methyl',
 'Trifloxystrobin',
 'X2.phenylphenol',
 'Zoxamide',
 'Diphenylamine',
 'Tebuconazole']

Insecticide :

In [16]:
classe_pesticides_df[classe_pesticides_df['typeEN'] == 'insecticide'].head(5)

importName                 name          statusEN   LMR  \
1                Aldrin               aldrin        prohibited  0.01   
13  Chlorantraniliprole  chlorantraniliprole        authorized  0.05   
17  Chlorpyrifos.methyl  chlorpyrifos-methyl        prohibited  0.05   
19            Coumaphos            coumaphos        prohibited  0.10   
20         Cyanofenphos         cyanofenphos  not yet assessed   NaN   

               familyEN       typeEN  
1        organochlorine  insecticide  
13  anthranilic diamide  insecticide  
17      organophosphate  insecticide  
19        thiophosphate  insecticide  
20      organophosphate  insecticide

In [17]:
insecticides = classe_pesticides_df[classe_pesticides_df['typeEN'] == 'insecticide'].importName.to_list()
insecticides.append('Diphenylamine')
insecticides.append('Methiocarb_total')
insecticides

['Aldrin',
 'Chlorantraniliprole',
 'Chlorpyrifos.methyl',
 'Coumaphos',
 'Cyanofenphos',
 'Cyflumetofen',
 'Cypermethrin',
 'Deet',
 'Dieldrin',
 'Etofenprox',
 'Fenoxycarb',
 'Fenpropathrin',
 'Flonicamid',
 'Flucythrinate',
 'Flupyradifurone',
 'Imidacloprid',
 'Indoxacarb',
 'Methoxyfenozide',
 'Phosmet',
 'Piperonylbutoxide',
 'Pirimicarb',
 'Spinosad',
 'Spirotetramat',
 'Tau.fluvalinate',
 'Tebufenozide',
 'Tetradifon',
 'Thiacloprid',
 'Diphenylamine',
 'Methiocarb_total']

Herbicide : 

In [18]:
classe_pesticides_df[classe_pesticides_df['typeEN'] == 'herbicide'].head(5)

importName           name    statusEN   LMR          familyEN  \
0       Aclonifen      aclonifen  authorized  0.05    diphenyl ether   
6     Benfluralin    benfluralin  authorized  0.05    dinitroaniline   
15  Chlorotoluron  chlorotoluron  authorized  0.05        phenylurea   
16   Chlorpropham   chlorpropham  prohibited  0.05         carbamate   
18      Clethodim      clethodim  authorized  0.05  cyclohexanedione   

       typeEN  
0   herbicide  
6   herbicide  
15  herbicide  
16  herbicide  
18  herbicide

In [19]:
herbicides = classe_pesticides_df[classe_pesticides_df['typeEN'] == 'herbicide'].importName.to_list()
herbicides

['Aclonifen',
 'Benfluralin',
 'Chlorotoluron',
 'Chlorpropham',
 'Clethodim',
 'Cycloate',
 'Diflufenican',
 'Ethofumesate',
 'Fluazifop',
 'Flufenacet',
 'Metamitron',
 'Methabenzthiazuron',
 'Metobromuron',
 'Metolachlor',
 'Nicosulfuron',
 'Pendimethalin',
 'Phenmedipham',
 'Propanil',
 'Propyzamide',
 'Prosulfocarb',
 'Prosulfuron',
 'Simazine',
 'Terbuthylazine',
 'Tri.allate',
 'Trifluralin']

Others :

In [20]:
classe_pesticides_df[(classe_pesticides_df['typeEN'] != 'fungicide') & 
                     (classe_pesticides_df['typeEN'] != 'insecticide') & 
                     (classe_pesticides_df['typeEN'] != 'herbicide')]

importName           name    statusEN   LMR  \
3           Anthraquinone  anthraquinone  prohibited  0.02   
34          Diphenylamine  diphenylamine  prohibited  0.05   
69       Methiocarb_total     methiocarb  prohibited  0.05   
86             Propargite     propargite  prohibited  0.05   
100          Tebuconazole   tebuconazole  authorized  0.05   
112  X6.benzylaminopurine  benzyladenine  authorized   NaN   
114                   NaN            NaN         NaN   NaN   

                            familyEN  \
3    polycyclic aromatic hydrocarbon   
34                             amine   
69                         carbamate   
86                               NaN   
100                         triazole   
112                       cytokinins   
114                              NaN   

                                           typeEN  
3                                  bird repellant  
34    growth regulator, fungicide and insecticide  
69   insecticide, molluscicide and bird repellent  
86                                      acaricide  
100                fungicide and growth regulator  
112                              growth regulator  
114                                           NaN

##### Choice to treat the others

Adding to the corresponding categories, stay in "other" ? What about the growth regulator ?

#### Add some columns

In [21]:
pesticides_df['qtt_fungicides'] = pesticides_df.apply(qtt_pesticides, args = fungicides, axis = 1)
pesticides_df['qtt_insecticides'] = pesticides_df.apply(qtt_pesticides, args = insecticides, axis = 1)
pesticides_df['qtt_herbicides'] = pesticides_df.apply(qtt_pesticides, args = herbicides, axis = 1)

In [22]:
pesticides_df['nb_fungicides'] = pesticides_df.apply(nb_pesticides, args = fungicides, axis = 1)
pesticides_df['nb_insecticides'] = pesticides_df.apply(nb_pesticides, args = insecticides, axis = 1)
pesticides_df['nb_herbicides'] = pesticides_df.apply(nb_pesticides, args = herbicides, axis = 1)

In [23]:
pesticides_df['lmr_fungicides'] = pesticides_df.apply(lmr_pesticides, args = fungicides, axis = 1)
pesticides_df['lmr_insecticides'] = pesticides_df.apply(lmr_pesticides, args = insecticides, axis = 1)
pesticides_df['lmr_herbicides'] = pesticides_df.apply(lmr_pesticides, args = herbicides, axis = 1)

In [24]:
pesticides_df.head(10)

REF....SUBSTANCE   PERIOD  Aldrin  Benalaxyl  Chlorpropham  Cyanofenphos  \
0            B0017  2017/P1     0.0     0.0000          0.00           0.0   
1            B0019  2017/P1     0.0     0.0188          0.00           0.0   
2            B0021  2017/P1     0.0     0.0711          0.00           0.0   
3            B0023  2017/P1     0.0     0.0294          0.00           0.0   
4            B0037  2017/P1     0.0     0.0000          0.01           0.0   
5            B0039  2017/P1     0.0     0.0000          0.00           0.0   
6            B0041  2017/P1     0.0     0.0000          0.00           0.0   
7            B0043  2017/P1     0.0     0.0000          0.00           0.0   
8            B0045  2017/P1     0.0     0.0000          0.00           0.0   
9            B0059  2017/P1     0.0     0.0000          0.00           0.0   

   Cypermethrin  Ethofumesate  Fludioxonil  Fluopyram  ...  Pentachloroanisol  \
0          0.00           0.0         0.00        0.0  ...                0.0   
1          0.00           0.0         0.00        0.0  ...                0.0   
2          0.00           0.0         0.00        0.0  ...                0.0   
3          0.00           0.0         0.00        0.0  ...                0.0   
4          0.00           0.0         0.00        0.0  ...                0.0   
5          0.01           0.0         0.00        0.0  ...                0.0   
6          0.00           0.0         0.00        0.0  ...                0.0   
7          0.00           0.0         0.01        0.0  ...                0.0   
8          0.00           0.0         0.00        0.0  ...                0.0   
9          0.00           0.0         0.00        0.0  ...                0.0   

   qtt_fungicides  qtt_insecticides  qtt_herbicides  nb_fungicides  \
0          0.0000            0.0000            0.00              0   
1          0.0188            0.0000            0.00              1   
2          0.0711            0.0000            0.00              1   
3          0.0294            0.0000            0.00              1   
4          0.0000            0.0000            0.01              0   
5          0.0000            0.0208            0.00              0   
6          0.0000            0.0000            0.00              0   
7          0.0100            0.0000            0.00              1   
8          0.0000            0.0000            0.00              0   
9          0.0145            0.0000            0.00              1   

   nb_insecticides  nb_herbicides  lmr_fungicides  lmr_insecticides  \
0                0              0               0                 0   
1                0              0               0                 0   
2                0              0               1                 0   
3                0              0               0                 0   
4                0              1               0                 0   
5                2              0               0                 0   
6                0              0               0                 0   
7                0              0               0                 0   
8                0              0               0                 0   
9                0              0               0                 0   

   lmr_herbicides  
0               0  
1               0  
2               0  
3               0  
4               0  
5               0  
6               0  
7               0  
8               0  
9               0  

[10 rows x 126 columns]

In [25]:
pesticides_df['year'] = pesticides_df['PERIOD'].apply(lambda x: x[:4])
pesticides_df['trimester'] = pesticides_df['PERIOD'].apply(lambda x: x[-1:])
pesticides_df.sample(10)

REF....SUBSTANCE   PERIOD  Aldrin  Benalaxyl  Chlorpropham  Cyanofenphos  \
400            B0093  2020/P2     0.0     0.1329        0.0000           0.0   
218            F0009  2019/P1     0.0     0.0000        0.0000           0.0   
379            F0061  2020/P1     0.0     0.0000        0.0000           0.0   
102            F0029  2018/P1     0.0     0.0000        0.0000           0.0   
307            F0041  2019/P3     0.0     0.0000        0.0000           0.0   
327            F0017  2019/P4     0.0     0.0000        0.0000           0.0   
69             B0105  2017/P4     0.0     0.0000        0.0137           0.0   
8              B0045  2017/P1     0.0     0.0000        0.0000           0.0   
207            B0069  2019/P1     0.0     0.0000        0.0000           0.0   
319            B0077  2019/P4     0.0     0.0423        0.0000           0.0   

     Cypermethrin  Ethofumesate  Fludioxonil  Fluopyram  ...  \
400           0.0           0.0       0.0000      0.000  ...   
218           0.0           0.0       0.0000      0.000  ...   
379           0.0           0.0       0.0000      0.015  ...   
102           0.0           0.0       0.0000      0.000  ...   
307           0.0           0.0       0.0000      0.000  ...   
327           0.0           0.0       0.0000      0.000  ...   
69            0.0           0.0       0.0000      0.000  ...   
8             0.0           0.0       0.0000      0.000  ...   
207           0.0           0.0       0.0121      0.000  ...   
319           0.0           0.0       0.0085      0.000  ...   

     qtt_insecticides  qtt_herbicides  nb_fungicides  nb_insecticides  \
400            0.0000          0.0039              2                0   
218            0.0000          0.0000              0                0   
379            0.0090          0.0185              6                2   
102            0.0000          0.0000              1                0   
307            0.0204          0.0000              2                1   
327            0.0000          0.0000              0                0   
69             0.0000          0.0805              0                0   
8              0.0000          0.0000              0                0   
207            0.0000          0.0109              6                0   
319            0.0000          0.0000              6                0   

     nb_herbicides  lmr_fungicides  lmr_insecticides  lmr_herbicides  year  \
400              1               1                 0               0  2020   
218              0               0                 0               0  2019   
379              4               0                 0               0  2020   
102              0               1                 0               0  2018   
307              0               0                 0               0  2019   
327              0               0                 0               0  2019   
69               3               0                 0               1  2017   
8                0               0                 0               0  2017   
207              2               1                 0               0  2019   
319              0               0                 0               0  2019   

     trimester  
400          2  
218          1  
379          1  
102          1  
307          3  
327          4  
69           4  
8            1  
207          1  
319          4  

[10 rows x 128 columns]

##### Alternate (smaller) dataset for pesticides

In [26]:
pesti_names = pesticides_df.columns.to_list()[1:-11]

In [27]:
small_pesticides_df = pesticides_df.drop(pesti_names, axis=1)
small_pesticides_df.head(5)

REF....SUBSTANCE  qtt_fungicides  qtt_insecticides  qtt_herbicides  \
0            B0017          0.0000               0.0            0.00   
1            B0019          0.0188               0.0            0.00   
2            B0021          0.0711               0.0            0.00   
3            B0023          0.0294               0.0            0.00   
4            B0037          0.0000               0.0            0.01   

   nb_fungicides  nb_insecticides  nb_herbicides  lmr_fungicides  \
0              0                0              0               0   
1              1                0              0               0   
2              1                0              0               1   
3              1                0              0               0   
4              0                0              1               0   

   lmr_insecticides  lmr_herbicides  year trimester  
0                 0               0  2017         1  
1                 0               0  2017         1  
2                 0               0  2017         1  
3                 0               0  2017         1  
4                 0               0  2017         1

##### Alternate (group by) dataset for pesticides

In [28]:
group_pesti_df = pesticides_df.groupby('REF....SUBSTANCE').agg(max)

In [29]:
group_pesti_df = group_pesti_df.loc[:,'lmr_fungicides' : 'lmr_herbicides']
group_pesti_df

lmr_fungicides  lmr_insecticides  lmr_herbicides
REF....SUBSTANCE                                                  
B0012                          1                 0               0
B0017                          1                 0               0
B0019                          0                 0               0
B0021                          1                 0               0
B0023                          0                 0               0
B0032                          1                 0               0
B0037                          1                 1               0
B0039                          1                 0               0
B0041                          1                 0               0
B0043                          1                 0               1
B0045                          1                 0               0
B0059                          1                 0               0
B0061                          0                 0               0
B0067                          1                 0               0
B0069                          1                 1               0
B0077                          1                 1               0
B0093                          1                 0               0
B0096                          0                 0               0
B0103                          1                 0               0
B0105                          1                 1               1
B0116                          0                 0               0
B0128                          0                 0               0
B0140                          0                 0               0
B0142                          1                 0               0
B0152                          1                 0               0
B0158                          1                 0               0
B0163                          1                 0               0
F0001                          1                 1               0
F0003                          1                 0               0
F0005                          0                 0               0
F0007                          0                 0               0
F0009                          0                 0               0
F0011                          0                 1               0
F0013                          0                 0               0
F0015                          1                 0               0
F0017                          0                 0               0
F0019                          1                 0               0
F0020                          1                 0               0
F0021                          1                 0               0
F0023                          0                 0               1
F0025                          0                 0               0
F0029                          1                 0               0
F0031                          1                 1               0
F0033                          0                 0               0
F0035                          0                 0               1
F0039                          1                 0               0
F0041                          0                 0               0
F0043                          0                 0               0
F0055                          1                 0               0
F0057                          1                 0               0
F0061                          1                 0               0
F0063                          1                 0               0
F0065                          0                 0               0
F0069                          0                 0               1
F0071                          1                 0               0
F0073                          1                 1               0
FE0001                         0                 0               0

##### Alternate df, with categorical

In [30]:
cat_pesti_df = group_pesti_df.copy(deep=True)

In [31]:
cat_pesti_df['cat'] = cat_pesti_df.apply(lambda x: (str(x['lmr_fungicides']) + 
                                                    str(x['lmr_insecticides']) + 
                                                    str(x['lmr_herbicides']))  , axis=1)

In [32]:
cat_pesti_df

lmr_fungicides  lmr_insecticides  lmr_herbicides  cat
REF....SUBSTANCE                                                       
B0012                          1                 0               0  100
B0017                          1                 0               0  100
B0019                          0                 0               0  000
B0021                          1                 0               0  100
B0023                          0                 0               0  000
B0032                          1                 0               0  100
B0037                          1                 1               0  110
B0039                          1                 0               0  100
B0041                          1                 0               0  100
B0043                          1                 0               1  101
B0045                          1                 0               0  100
B0059                          1                 0               0  100
B0061                          0                 0               0  000
B0067                          1                 0               0  100
B0069                          1                 1               0  110
B0077                          1                 1               0  110
B0093                          1                 0               0  100
B0096                          0                 0               0  000
B0103                          1                 0               0  100
B0105                          1                 1               1  111
B0116                          0                 0               0  000
B0128                          0                 0               0  000
B0140                          0                 0               0  000
B0142                          1                 0               0  100
B0152                          1                 0               0  100
B0158                          1                 0               0  100
B0163                          1                 0               0  100
F0001                          1                 1               0  110
F0003                          1                 0               0  100
F0005                          0                 0               0  000
F0007                          0                 0               0  000
F0009                          0                 0               0  000
F0011                          0                 1               0  010
F0013                          0                 0               0  000
F0015                          1                 0               0  100
F0017                          0                 0               0  000
F0019                          1                 0               0  100
F0020                          1                 0               0  100
F0021                          1                 0               0  100
F0023                          0                 0               1  001
F0025                          0                 0               0  000
F0029                          1                 0               0  100
F0031                          1                 1               0  110
F0033                          0                 0               0  000
F0035                          0                 0               1  001
F0039                          1                 0               0  100
F0041                          0                 0               0  000
F0043                          0                 0               0  000
F0055                          1                 0               0  100
F0057                          1                 0               0  100
F0061                          1                 0               0  100
F0063                          1                 0               0  100
F0065                          0                 0               0  000
F0069                          0                 0        

In [33]:
cat_pesti_df['cat'] = cat_pesti_df['cat'].astype('category')
cat_pesti_df['cat_label'] = cat_pesti_df['cat'].cat.codes

In [34]:
labels = cat_pesti_df.set_index('cat_label')['cat'].to_dict()
labels

{3: '100', 0: '000', 5: '110', 4: '101', 6: '111', 2: '010', 1: '001'}

In [35]:
cat_pesti_df = cat_pesti_df.drop(columns=['lmr_fungicides', 
                                          'lmr_insecticides', 
                                          'lmr_herbicides', 
                                          'cat'])

In [36]:
cat_pesti_df

cat_label
REF....SUBSTANCE           
B0012                     3
B0017                     3
B0019                     0
B0021                     3
B0023                     0
B0032                     3
B0037                     5
B0039                     3
B0041                     3
B0043                     4
B0045                     3
B0059                     3
B0061                     0
B0067                     3
B0069                     5
B0077                     5
B0093                     3
B0096                     0
B0103                     3
B0105                     6
B0116                     0
B0128                     0
B0140                     0
B0142                     3
B0152                     3
B0158                     3
B0163                     3
F0001                     5
F0003                     3
F0005                     0
F0007                     0
F0009                     0
F0011                     2
F0013                     0
F0015                     3
F0017                     0
F0019                     3
F0020                     3
F0021                     3
F0023                     1
F0025                     0
F0029                     3
F0031                     5
F0033                     0
F0035                     1
F0039                     3
F0041                     0
F0043                     0
F0055                     3
F0057                     3
F0061                     3
F0063                     3
F0065                     0
F0069                     1
F0071                     3
F0073                     5
FE0001                    0

### Sites

##### Getting the area for each zone

In [37]:
distances_df.head(5)

Site     polyID  classCLC         dist
0  F0069   FR-88090       231  1183.022362
1  F0069  FR-131773       242   289.926324
2  F0069   FR-46167       211   221.344153
3  F0069   FR-11047       112  1038.383069
4  F0069  FR-195343       311   724.045683

In [38]:
surfaces_df.head(5)

Site  111           112           121            122  123  124  131  132  \
0  F0069  0.0  1.064122e+06  3.587087e+05       0.000000  0.0  0.0  0.0    0   
1  F0025  0.0  0.000000e+00  0.000000e+00  412752.692383  0.0  0.0  0.0    0   
2  F0027  0.0  2.741746e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
3  B0128  0.0  2.470089e+06  4.053755e+06  497496.251770  0.0  0.0  0.0    0   
4  F0023  0.0  0.000000e+00  2.559254e+05       0.000000  0.0  0.0  0.0    0   

   133  ...  411  412  421  422  423  511            512  521  522  523  
0  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
1  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
2  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
3  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
4  0.0  ...    0  0.0    0    0  0.0  0.0  284728.392578    0    0  0.0  

[5 rows x 45 columns]

In [39]:
# surfaces_df['country'] = surfaces_df['Site'].apply(lambda x: 'FR' if x.startswith('F') else 'BE')
# surfaces_df['country'].value_counts()

In [40]:
sites_df = distances_df.copy(deep=True)

In [41]:
sites_df['surface'] = sites_df.apply(get_surface, axis = 1)

In [42]:
sites_df.sample(10)

Site     polyID  classCLC         dist       surface
21   F0025   FR-30657       122  1177.944218  4.127527e+05
282  B0093     BE-543       112   346.805188  5.321409e+05
167  F0067   FR-29418       121    17.234548  2.077216e+06
154  F0075   FR-29212       121   344.517523  4.422800e+05
264  B0059    BE-3583       112  1271.934537  8.684684e+05
534  F0013  FR-121947       242  1208.917362  3.958112e+05
367  F0031   FR-22828       112  1398.482972  3.422013e+05
193  B0140    BE-4846       141   153.023167  2.016853e+06
272  B0059   BE-16800       242  1054.380546  1.382229e+06
184  B0140      BE-22       111  1019.756755  1.011569e+06

In [43]:
sites_df['country'] = sites_df['polyID'].apply(lambda x: 'FR' if x.startswith('FR') else 'BE')
sites_df.sample(10)

Site     polyID  classCLC         dist       surface country
243  B0039    BE-3868       112     0.000000  2.951602e+06      BE
53   F0071  FR-213676       312   335.968443  4.684407e+05      FR
594  B0069    BE-1392       231   381.929259  2.521990e+05      BE
537  F0013   FR-81265       231     0.000000  2.151188e+06      FR
365  F0031  FR-103934       231     0.000000  4.599038e+06      FR
25   F0027   FR-15065       112  1052.832822  2.741746e+05      FR
435  B0105    BE-2397       112   804.818038  6.810172e+05      BE
320  F0017   FR-48582       211   689.236741  9.580098e+05      FR
186  B0140    BE-2572       112   499.793836  2.091070e+06      BE
264  B0059    BE-3583       112  1271.934537  8.684684e+05      BE

In [44]:
sites_df['country'].value_counts()

FR    335
BE    297
Name: country, dtype: int64

In [45]:
sites_df = sites_df.drop(['polyID'], axis=1)
sites_df.sample(5)

Site  classCLC         dist       surface country
277  B0059       243  1377.335085  2.254571e+05      BE
591  F0007       242   430.140176  1.546218e+06      FR
170  F0067       313   794.331161  2.955444e+05      FR
312  F0057       211   587.063036  2.953287e+06      FR
50   F0071       311   860.707116  5.607325e+05      FR

##### Get one dataset with : beehives and area/distance for each zone

In [46]:
surfaces_df

Site  111           112           121            122  123  124  131  132  \
0   F0069  0.0  1.064122e+06  3.587087e+05       0.000000  0.0  0.0  0.0    0   
1   F0025  0.0  0.000000e+00  0.000000e+00  412752.692383  0.0  0.0  0.0    0   
2   F0027  0.0  2.741746e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
3   B0128  0.0  2.470089e+06  4.053755e+06  497496.251770  0.0  0.0  0.0    0   
4   F0023  0.0  0.000000e+00  2.559254e+05       0.000000  0.0  0.0  0.0    0   
..    ...  ...           ...           ...            ...  ...  ...  ...  ...   
58  F0007  0.0  0.000000e+00  0.000000e+00       0.000000  0.0  0.0  0.0    0   
59  B0069  0.0  8.704243e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
60  B0152  0.0  4.575279e+06  5.726391e+05  201434.000214  0.0  0.0  0.0    0   
61  F0003  0.0  3.425116e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
62  B0158  0.0  1.467398e+05  0.000000e+00  318178.187439  0.0  0.0  0.0    0   

    133  ...  411  412  421  422  423  511            512  521  522  523  
0   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
1   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
2   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
3   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
4   0.0  ...    0  0.0    0    0  0.0  0.0  284728.392578    0    0  0.0  
..  ...  ...  ...  ...  ...  ...  ...  ...            ...  ...  ...  ...  
58  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
59  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
60  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
61  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
62  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  

[63 rows x 45 columns]

In [47]:
classCLC = surfaces_df.columns.to_list()[1:]

In [48]:
by_sites_df = surfaces_df.copy(deep=True)
by_sites_df

Site  111           112           121            122  123  124  131  132  \
0   F0069  0.0  1.064122e+06  3.587087e+05       0.000000  0.0  0.0  0.0    0   
1   F0025  0.0  0.000000e+00  0.000000e+00  412752.692383  0.0  0.0  0.0    0   
2   F0027  0.0  2.741746e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
3   B0128  0.0  2.470089e+06  4.053755e+06  497496.251770  0.0  0.0  0.0    0   
4   F0023  0.0  0.000000e+00  2.559254e+05       0.000000  0.0  0.0  0.0    0   
..    ...  ...           ...           ...            ...  ...  ...  ...  ...   
58  F0007  0.0  0.000000e+00  0.000000e+00       0.000000  0.0  0.0  0.0    0   
59  B0069  0.0  8.704243e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
60  B0152  0.0  4.575279e+06  5.726391e+05  201434.000214  0.0  0.0  0.0    0   
61  F0003  0.0  3.425116e+05  0.000000e+00       0.000000  0.0  0.0  0.0    0   
62  B0158  0.0  1.467398e+05  0.000000e+00  318178.187439  0.0  0.0  0.0    0   

    133  ...  411  412  421  422  423  511            512  521  522  523  
0   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
1   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
2   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
3   0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
4   0.0  ...    0  0.0    0    0  0.0  0.0  284728.392578    0    0  0.0  
..  ...  ...  ...  ...  ...  ...  ...  ...            ...  ...  ...  ...  
58  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
59  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
60  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
61  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  
62  0.0  ...    0  0.0    0    0  0.0  0.0       0.000000    0    0  0.0  

[63 rows x 45 columns]

In [49]:
for CLC in classCLC:
    by_sites_df[CLC] = by_sites_df.apply(distance_area, args = (CLC,), axis = 1)

In [50]:
by_sites_df

Site  111           112           121         122  123  124  131  132  \
0   F0069  0.0  1.442780e+03  3.587087e+05    0.000000  0.0  0.0  0.0  0.0   
1   F0025  0.0  0.000000e+00  0.000000e+00  700.801763  0.0  0.0  0.0  0.0   
2   F0027  0.0  2.604161e+02  0.000000e+00    0.000000  0.0  0.0  0.0  0.0   
3   B0128  0.0  2.412962e+03  4.053755e+06  523.436355  0.0  0.0  0.0  0.0   
4   F0023  0.0  0.000000e+00  7.907522e+02    0.000000  0.0  0.0  0.0  0.0   
..    ...  ...           ...           ...         ...  ...  ...  ...  ...   
58  F0007  0.0  0.000000e+00  0.000000e+00    0.000000  0.0  0.0  0.0  0.0   
59  B0069  0.0  7.328884e+03  0.000000e+00    0.000000  0.0  0.0  0.0  0.0   
60  B0152  0.0  4.575279e+06  5.227880e+02  421.402885  0.0  0.0  0.0  0.0   
61  F0003  0.0  3.859675e+02  0.000000e+00    0.000000  0.0  0.0  0.0  0.0   
62  B0158  0.0  1.574508e+02  0.000000e+00  392.706036  0.0  0.0  0.0  0.0   

    133  ...  411  412  421  422  423  511         512  521  522  523  
0   0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
1   0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
2   0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
3   0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
4   0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0  463.344615  0.0  0.0  0.0  
..  ...  ...  ...  ...  ...  ...  ...  ...         ...  ...  ...  ...  
58  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
59  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
60  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
61  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  
62  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  

[63 rows x 45 columns]

check if correct

In [51]:
distance = distances_df[(distances_df['Site'] == 'F0023') & 
                        (distances_df['classCLC'] == 512)]['dist'].values
distance

array([614.50674791])

In [52]:
area = surfaces_df[surfaces_df['Site'] == 'F0023']['512'].values[0]
area

284728.392578125

In [53]:
area/distance[0]

463.3446150839585

##### Drop some colomns that don't contain data

In [54]:
by_sites_df = by_sites_df.drop(['132','212','213','222', '223', '241', '244',
              '321', '323', '332', '333', '334', '335', '411',
              '421', '422', '521', '522', '523'], axis=1)

### Merging the datasets into two

3 datasets to 2 : sites + (heavy metal || pesticides)

##### Heavy Metal

##### Pesticides

full

In [55]:
pesticides_df.sample(5)

REF....SUBSTANCE   PERIOD  Aldrin  Benalaxyl  Chlorpropham  Cyanofenphos  \
213            B0152  2019/P1     0.0        0.0           0.0           0.0   
455            B0152  2020/P3     0.0        0.0           0.0           0.0   
488            B0067  2020/P4     0.0        0.0           0.0           0.0   
225            F0021  2019/P1     0.0        0.0           0.0           0.0   
491            B0093  2020/P4     0.0        0.0           0.0           0.0   

     Cypermethrin  Ethofumesate  Fludioxonil  Fluopyram  ...  \
213           0.0           0.0          0.0     0.0000  ...   
455           0.0           0.0          0.0     0.0000  ...   
488           0.0           0.0          0.0     0.0000  ...   
225           0.0           0.0          0.0     0.0000  ...   
491           0.0           0.0          0.0     0.0064  ...   

     qtt_insecticides  qtt_herbicides  nb_fungicides  nb_insecticides  \
213               0.0             0.0              0                0   
455               0.0             0.0              0                0   
488               0.0             0.0              2                0   
225               0.0             0.0              1                0   
491               0.0             0.0              2                0   

     nb_herbicides  lmr_fungicides  lmr_insecticides  lmr_herbicides  year  \
213              0               0                 0               0  2019   
455              0               0                 0               0  2020   
488              0               0                 0               0  2020   
225              0               0                 0               0  2019   
491              0               0                 0               0  2020   

     trimester  
213          1  
455          3  
488          4  
225          1  
491          4  

[5 rows x 128 columns]

In [56]:
small_pesticides_df.sample(5)

REF....SUBSTANCE  qtt_fungicides  qtt_insecticides  qtt_herbicides  \
413            F0011          0.0000               0.0          0.0000   
406            B0152          0.0230               0.0          0.0000   
411            F0007          0.0000               0.0          0.0000   
99             F0021          0.1264               0.0          0.0000   
514            F0057          0.0000               0.0          0.0204   

     nb_fungicides  nb_insecticides  nb_herbicides  lmr_fungicides  \
413              0                0              0               0   
406              1                0              0               0   
411              0                0              0               0   
99               5                0              0               0   
514              0                0              1               0   

     lmr_insecticides  lmr_herbicides  year trimester  
413                 0               0  2020         2  
406                 0               0  2020         2  
411                 0               0  2020         2  
99                  0               0  2018         1  
514                 0               0  2020         4

In [57]:
group_pesti_df.sample(5)

lmr_fungicides  lmr_insecticides  lmr_herbicides
REF....SUBSTANCE                                                  
B0043                          1                 0               1
F0029                          1                 0               0
B0045                          1                 0               0
B0061                          0                 0               0
B0103                          1                 0               0

In [58]:
cat_pesti_df.sample(5)

cat_label
REF....SUBSTANCE           
B0105                     6
F0063                     3
F0043                     0
F0020                     3
B0116                     0

##### sites

In [59]:
sites_df.sample(5)

Site  classCLC        dist       surface country
258  F0035       312  888.468489  9.134774e+05      FR
568  F0009       242  985.010031  1.979240e+06      FR
129  F0019       142  353.879794  1.748277e+06      FR
365  F0031       231    0.000000  4.599038e+06      FR
490  F0045       512  634.904248  3.375305e+05      FR

In [60]:
by_sites_df.sample(5)

Site  111          112           121         122  123  124  \
8   F0001  0.0   421.795587  0.000000e+00    0.000000  0.0  0.0   
15  F0041  0.0     0.000000  3.383666e+06  979.946280  0.0  0.0   
54  F0015  0.0     0.000000  0.000000e+00    0.000000  0.0  0.0   
13  F0019  0.0  3367.677891  4.276874e+02  334.489944  0.0  0.0   
53  F0013  0.0  2320.995757  0.000000e+00    0.000000  0.0  0.0   

              131  133         141  ...           311          312  \
8        0.000000  0.0    0.000000  ...  3.182421e+06     0.000000   
15       0.000000  0.0  506.157789  ...  0.000000e+00     0.000000   
54  309985.414551  0.0    0.000000  ...  4.969729e+04  2610.097612   
13       0.000000  0.0    0.000000  ...  7.228632e+02     0.000000   
53       0.000000  0.0    0.000000  ...  2.226354e+03  1645.072615   

           313         322       324  331  412  423  511  512  
8   314.918152    0.000000   0.00000  0.0  0.0  0.0  0.0  0.0  
15    0.000000    0.000000   0.00000  0.0  0.0  0.0  0.0  0.0  
54    0.000000    0.000000   0.00000  0.0  0.0  0.0  0.0  0.0  
13    0.000000    0.000000   0.00000  0.0  0.0  0.0  0.0  0.0  
53    0.000000  485.797839  89.28406  0.0  0.0  0.0  0.0  0.0  

[5 rows x 26 columns]

#### Merge Pesticides

In [61]:
cat_pesti_df.index.name

'REF....SUBSTANCE'

In [62]:
cat_pesti_df = cat_pesti_df.reset_index()

In [63]:
cat_pesti_df = cat_pesti_df.rename(columns={"REF....SUBSTANCE": "Site"})

In [64]:
cat_pesti_df.head(3)

Site  cat_label
0  B0012          3
1  B0017          3
2  B0019          0

In [65]:
by_sites_df.head(3)

Site  111          112            121         122  123  124  131  133  \
0  F0069  0.0  1442.780424  358708.730957    0.000000  0.0  0.0  0.0  0.0   
1  F0025  0.0     0.000000       0.000000  700.801763  0.0  0.0  0.0  0.0   
2  F0027  0.0   260.416070       0.000000    0.000000  0.0  0.0  0.0  0.0   

   141  ...           311  312  313  322         324  331  412  423  511  512  
0  0.0  ...    352.189664  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  0.0  0.0  
1  0.0  ...  24001.488078  0.0  0.0  0.0  926.217913  0.0  0.0  0.0  0.0  0.0  
2  0.0  ...  11215.516811  0.0  0.0  0.0    0.000000  0.0  0.0  0.0  0.0  0.0  

[3 rows x 26 columns]

In [66]:
merged_pesti = pd.merge(cat_pesti_df, by_sites_df, on=["Site"])

In [67]:
merged_pesti

Site  cat_label           111           112           121          122  \
0    B0012          3    209.227976  6.653099e+03  1.764701e+04     0.000000   
1    B0017          3      0.000000  7.607874e+03  2.319998e+00     0.000000   
2    B0019          0      0.000000  0.000000e+00  0.000000e+00     0.000000   
3    B0021          3      0.000000  0.000000e+00  0.000000e+00     0.000000   
4    B0023          0      0.000000  0.000000e+00  0.000000e+00     0.000000   
5    B0032          3      0.000000  2.180353e+05  3.227128e+03     0.000000   
6    B0037          5      0.000000  1.274631e+04  0.000000e+00     0.000000   
7    B0039          3   3098.057557  2.951602e+06  0.000000e+00     0.000000   
8    B0041          3    391.077976  2.904582e+06  1.541424e+02     0.000000   
9    B0043          4      0.000000  5.390840e+02  0.000000e+00     0.000000   
10   B0045          3      0.000000  4.456532e+03  0.000000e+00  1032.436781   
11   B0059          3      0.000000  7.916798e+02  5.337917e+02   468.730666   
12   B0061          0      0.000000  1.225678e+02  0.000000e+00     0.000000   
13   B0067          3      0.000000  5.171075e+04  0.000000e+00     0.000000   
14   B0069          5      0.000000  7.328884e+03  0.000000e+00     0.000000   
15   B0077          5      0.000000  4.108273e+02  0.000000e+00    87.748165   
16   B0093          3      0.000000  1.534409e+03  0.000000e+00     0.000000   
17   B0096          0      0.000000  5.206697e+06  0.000000e+00     0.000000   
18   B0103          3      0.000000  4.401865e+02  0.000000e+00     0.000000   
19   B0105          6      0.000000  1.102039e+03  1.153518e+05     0.000000   
20   B0116          0      0.000000  4.836129e+06  3.770519e+02   546.163532   
21   B0128          0      0.000000  2.412962e+03  4.053755e+06   523.436355   
22   B0140          0   1416.095618  2.959663e+03  1.258021e+06  1967.770812   
23   B0142          3      0.000000  6.808838e+03  6.009481e+03  1078.987269   
24   B0152          3      0.000000  4.575279e+06  5.227880e+02   421.402885   
25   B0158          3      0.000000  1.574508e+02  0.000000e+00   392.706036   
26   B0163          3  15107.362039  5.058683e+04  1.616543e+03    19.130127   
27   F0001          5      0.000000  4.217956e+02  0.000000e+00     0.000000   
28   F0003          3      0.000000  3.859675e+02  0.000000e+00     0.000000   
29   F0005          0      0.000000  5.240466e+02  0.000000e+00     0.000000   
30   F0007          0      0.000000  0.000000e+00  0.000000e+00     0.000000   
31   F0009          0      0.000000  0.000000e+00  0.000000e+00    65.325567   
32   F0011          2      0.000000  7.531979e+02  0.000000e+00     0.000000   
33   F0013          0      0.000000  2.320996e+03  0.000000e+00     0.000000   
34   F0015          3      0.000000  0.000000e+00  0.000000e+00     0.000000   
35   F0017          0      0.000000  2.366620e+02  0.000000e+00     0.000000   
36   F0019          3      0.000000  3.367678e+03  4.276874e+02   334.489944   
37   F0020          3      0.000000  1.029865e+03  0.000000e+00     0.000000   
38   F0021          3      0.000000  1.875282e+04  0.000000e+00     0.000000   
39   F0023          1      0.000000  0.000000e+00  7.907522e+02     0.000000   
40   F0025          0      0.000000  0.000000e+00  0.000000e+00   700.801763   
41   F0029          3      0.000000  2.793964e+02  0.000000e+00     0.000000   
42   F0031          5      0.000000  2.922115e+02  0.000000e+00     0.000000   
43   F0033          0      0.000000  2.201997e+01  0.000000e+00     0.000000   
44   F0035          1      0.000000  0.000000e+00  0.000000e+00     0.000000   
45   F0039          3      0.000000  3.274455e+01  4.708634e+02   200.632244   
46   F0041          0      0.000000  0.000000e+00  3.383666e+06   979.946280   
47   F0043          0      0.000000  1.502358e+03  8.333661e+02     0.000000   
48   F0055          3      0.000000  1.313065e+03  2.656270e+03     0.000000   
49   F

### Final Cleaning

##### HM

In [68]:
# merged_hm.to_csv('merged_hm.csv')

##### Pesticides

In [69]:
merged_pesti = merged_pesti.drop(['Site'], axis=1)

In [70]:
merged_pesti.sample(5)

cat_label         111           112            121          122  123  124  \
24          3     0.00000  4.575279e+06     522.788005   421.402885  0.0  0.0   
52          0  1967.70498  3.352559e+06    1416.685638  2509.438121  0.0  0.0   
23          3     0.00000  6.808838e+03    6009.480699  1078.987269  0.0  0.0   
19          6     0.00000  1.102039e+03  115351.826583     0.000000  0.0  0.0   
2           0     0.00000  0.000000e+00       0.000000     0.000000  0.0  0.0   

    131  133         141  ...         311           312          313  \
24  0.0  0.0    0.000000  ...    0.000000  0.000000e+00     0.000000   
52  0.0  0.0  191.539631  ...    0.000000  0.000000e+00     0.000000   
23  0.0  0.0    0.000000  ...    0.000000  0.000000e+00     0.000000   
19  0.0  0.0    0.000000  ...    0.000000  0.000000e+00     0.000000   
2   0.0  0.0    0.000000  ...  196.193589  5.146666e+06  1701.143283   

           322  324  331         412  423  511  512  
24  451.314675  0.0  0.0    0.000000  0.0  0.0  0.0  
52    0.000000  0.0  0.0    0.000000  0.0  0.0  0.0  
23    0.000000  0.0  0.0    0.000000  0.0  0.0  0.0  
19    0.000000  0.0  0.0    0.000000  0.0  0.0  0.0  
2     0.000000  0.0  0.0  818.792332  0.0  0.0  0.0  

[5 rows x 26 columns]

In [71]:
merged_pesti.to_csv('merged_pesti.csv')

# Questions to ask

- When NaN => information irrelevant or not analysed?
- Difference France/Belgium?
- Is the fact that an insecticide is unauthorized relevant? (=> not the same law in France/Belgium)
- Or do we have to show that even banned pesticides are still found?
- How do we treat other categorical pesticides?
- How to increase the number of datas?

### Bo-bo-bonus

- FR  :  36 beehives for 335 zones
- BE  :  27 beehives for 297 zones

Here the CLC zone are left separate, could group some to test